lib/std/fs/File.zig

```diff
@@ -1149,13 +1149,23 @@ pub fn readToEndAllocOptions(
     comptime alignment: u29,
     comptime optional_sentinel: ?u8,
 ) !(if (optional_sentinel) |s| [:s]align(alignment) u8 else []align(alignment) u8) {
-    // If no size hint is provided fall back to the size=0 code path
-    const size = size_hint orelse 0;
+    var size: usize = undefined;
+    if (size_hint) |hint| {
+        size = hint;
+    } else {
```

Comment:
I feel like the whole `size_hint` is not necessary anymore, why would someone give a size_hint that is lower than the full file size or something higher than it?

Since this function always reads a file to the end it should just allocate memory according to the size of the file.

Reply:
(But I wouldn't want to introduce a breaking change, especially without a discussion around it first)

Reply:
> I feel like the whole size_hint is not necessary anymore

According to the comments made by @squeek502 `size_hint` is valuable as it allows the caller to not do two calls to `stat()` so in the case of `fmtPathFile` which already calls `File.stat()` it can directly pass in the size of the file, instead of delegating the buffer size to the `readSourceFileToEndAlloc` and then to `readToEndAllocOptions` functions.